components/scream/cime_config/buildnml

```diff
@@ -47,52 +48,144 @@ def do_cime_vars(entry, case):
+                        all_match = False
+                        break
+
+                if all_match:
+                    orig_child = child_values[child_name]
+                    orig_child.text = do_cime_vars(child_val, case)
+
+                children_to_remove.append(child)
+
             else:
-                segments[-1] += entry[idx]
-
-        expect(idx < len(entry), "Switch statement parse error in string '{}'".format(entry))
-        end_switch_idx = idx + 1
-        best_val = None
-        for segment in segments:
-            components = segment.split("=>", 1)
-            if (len(components) == 1 or '<' in components[0]):
-                if best_val is None:
-                    best_val = do_switches(segment).strip()
+                expect(child_name not in child_values, "Repeat of child {} without selectors".format(child_name))
+                child_values[child_name] = child
+                child.text = do_cime_vars(child_val, case)
+
+        else:
+            evaluate_selectors(child, case, ez_selectors)
+
+    for child_to_remove in children_to_remove:
+        element.remove(child_to_remove)
+
+###############################################################################
+def create_raw_xml_file(case, caseroot):
+###############################################################################
+    src = os.path.join(case.get_value("SRCROOT"), "components/scream/cime_config/namelist_defaults_scream.xml")
+
+    raw_xml_file = os.path.join(caseroot, "namelist_scream.xml")
+    if os.path.exists(raw_xml_file):
+        print("{} already exists, will not overwrite. Remove to regenerate".format(raw_xml_file))
+    else:
+        with open(src, "r") as fd:
+            tree = ET.parse(fd)
+            root = tree.getroot()
+
+        selectors = None
+        selectors_elem = None
+        for child in root:
+            if selectors is None:
+                expect(child.tag == "selectors", "Unexpected child.tag {}".format(child.tag))
+                selectors_elem = child
+
+                selectors = {}
+                for selector in child:
+                    expect(selector.tag == "selector", "Expected selector tag, not {}".format(selector.tag))
+
+                    selector_name  = selector.attrib["name"]
+                    selector_env   = selector.attrib["case_env"]
+                    if "regex" in selector.attrib:
+                        selector_regex = selector.attrib["regex"]
+                    else:
+                        selector_regex = ".*" # Just grab the whole thing
+
+                    selectors[selector_name] = (selector_env, selector_regex)
+
             else:
-                compare = re.compile(components[0].strip())
-                if compare.match(value):
-                    best_val = do_switches(components[-1]).strip()
-                    break
+                expect(child.tag == "file", "Unexpected child.tag {}".format(child.tag))
```

Comment:
This only works now that we have two things to parse (an input.yaml file and a namelist.nl file) in the defaults. I suspect this will change very soon. But if you'd rather keep this check here for now, maybe add a comment warning, that this is only the case now, and might change in the future?

Reply:
I'm not sure I understand the problem here. `namelist_defaults_scream.xml` is used to produce program inputs will which always be in the form of one or more files. Currently, every major organizational block in this XML (with the exception of the easy selectors) should be defining an input file.

Reply:
I forgot to follow up on this. If I'm understanding you right, buildnml is already doing what you described. There are two expected attributes for the `file` element, `name` and `format`. The `name` can be something like `data/scream_input.yaml`, which tells buildnml where to put the file relative to RUNDIR.